coderd/database/foreign_key_constraint.go

```diff
@@ -0,0 +1,49 @@
+	ForeignKeyWorkspaceBuildParametersWorkspaceBuildIDFkey     ForeignKeyConstraint = "workspace_build_parameters_workspace_build_id_fkey"     // ALTER TABLE ONLY workspace_build_parameters ADD CONSTRAINT workspace_build_parameters_workspace_build_id_fkey FOREIGN KEY (workspace_build_id) REFERENCES workspace_builds(id) ON DELETE CASCADE;
+	ForeignKeyWorkspaceBuildsJobIDFkey                         ForeignKeyConstraint = "workspace_builds_job_id_fkey"                           // ALTER TABLE ONLY workspace_builds ADD CONSTRAINT workspace_builds_job_id_fkey FOREIGN KEY (job_id) REFERENCES provisioner_jobs(id) ON DELETE CASCADE;
+	ForeignKeyWorkspaceBuildsTemplateVersionIDFkey             ForeignKeyConstraint = "workspace_builds_template_version_id_fkey"              // ALTER TABLE ONLY workspace_builds ADD CONSTRAINT workspace_builds_template_version_id_fkey FOREIGN KEY (template_version_id) REFERENCES template_versions(id) ON DELETE CASCADE;
+	ForeignKeyWorkspaceBuildsWorkspaceIDFkey                   ForeignKeyConstraint = "workspace_builds_workspace_id_fkey"                     // ALTER TABLE ONLY workspace_builds ADD CONSTRAINT workspace_builds_workspace_id_fkey FOREIGN KEY (workspace_id) REFERENCES workspaces(id) ON DELETE CASCADE;
+	ForeignKeyWorkspaceResourceMetadataWorkspaceResourceIDFkey ForeignKeyConstraint = "workspace_resource_metadata_workspace_resource_id_fkey" // ALTER TABLE ONLY workspace_resource_metadata ADD CONSTRAINT workspace_resource_metadata_workspace_resource_id_fkey FOREIGN KEY (workspace_resource_id) REFERENCES workspace_resources(id) ON DELETE CASCADE;
+	ForeignKeyWorkspaceResourcesJobIDFkey                      ForeignKeyConstraint = "workspace_resources_job_id_fkey"                        // ALTER TABLE ONLY workspace_resources ADD CONSTRAINT workspace_resources_job_id_fkey FOREIGN KEY (job_id) REFERENCES provisioner_jobs(id) ON DELETE CASCADE;
+	ForeignKeyWorkspacesOrganizationIDFkey                     ForeignKeyConstraint = "workspaces_organization_id_fkey"                        // ALTER TABLE ONLY workspaces ADD CONSTRAINT workspaces_organization_id_fkey FOREIGN KEY (organization_id) REFERENCES organizations(id) ON DELETE RESTRICT;
+	ForeignKeyWorkspacesOwnerIDFkey                            ForeignKeyConstraint = "workspaces_owner_id_fkey"                               // ALTER TABLE ONLY workspaces ADD CONSTRAINT workspaces_owner_id_fkey FOREIGN KEY (owner_id) REFERENCES users(id) ON DELETE RESTRICT;
+	ForeignKeyWorkspacesTemplateIDFkey                         ForeignKeyConstraint = "workspaces_template_id_fkey"                            // ALTER TABLE ONLY workspaces ADD CONSTRAINT workspaces_template_id_fkey FOREIGN KEY (template_id) REFERENCES templates(id) ON DELETE RESTRICT;
```

Comment:
You could update `nameFromSnakeCase` with `uuid => UUID` and `fkey => [empty string]`. More Go-ish and removes duplication. calling `fkey` `FKey` is another option, but `Fkey` feels wrong 😄

Reply:
Updated with a few other changes. 👍

Reply:
Very often I get confused with snakeCase where it comes to phrases like `FoobarIDUUID` :) Should it be `FoobarIdUUID`, or `FoobarIDUuid`.

Reply:
From https://zchee.github.io/golang-wiki/CodeReviewComments/#initialisms:

> For identifiers with multiple initialized "words", use for example "xmlHTTPRequest" or "XMLHTTPRequest".

Reply:
Also: https://github.com/golang/lint/blob/master/lint.go#L770-L809 (this may be outdated, but these are what used to be a part of `golint`).